core/src/test/java/io/confluent/rest/ConnectionDurationFilterTest.java

```diff
@@ -0,0 +1,225 @@
+/*
+ * Copyright 2024 Confluent Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package io.confluent.rest;
+
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.junit.jupiter.api.Assertions.assertTrue;
+
+import com.google.common.collect.ImmutableMap;
+import java.net.URI;
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.List;
+import java.util.Map;
+import javax.servlet.http.HttpServletRequest;
+import javax.ws.rs.GET;
+import javax.ws.rs.Path;
+import javax.ws.rs.core.Configurable;
+import javax.ws.rs.core.Context;
+import javax.ws.rs.core.Response.Status;
+import javax.ws.rs.core.UriBuilder;
+import okhttp3.OkHttpClient;
+import okhttp3.Request;
+import okhttp3.Response;
+import org.apache.http.client.methods.CloseableHttpResponse;
+import org.apache.http.client.methods.HttpGet;
+import org.apache.http.impl.client.CloseableHttpClient;
+import org.apache.http.impl.client.HttpClients;
+import org.apache.http.impl.conn.BasicHttpClientConnectionManager;
+import org.apache.http.impl.conn.PoolingHttpClientConnectionManager;
+import org.apache.http.util.EntityUtils;
+import org.eclipse.jetty.server.Server;
+import org.junit.jupiter.api.AfterAll;
+import org.junit.jupiter.api.BeforeAll;
+import org.junit.jupiter.api.Test;
+
+
+/*
+ *  Note on client support --
+ *  The Apache HttpClient library v4.x does not support HTTP/2.0, so we use
+ *  OkHttp for HTTP/2.0 tests.
+ */
+public class ConnectionDurationFilterTest {
```

Comment:
I don't see a test where HTTP requests fail as long running TCP connection closes unexpectedly. Is that going to be added?

Reply:
Interestingly, the HTTP requests don't fail if the connection closes explicitly. 
Both apache http client as well as okhttp client are smart enough to see that the TCP connection is closed and they attempt to open a new connection under the hood.  That's really showing that the "graceful termination" actually worked. 

In an earlier implementation attempt for this feature, I tried to close the network socket (by just calling `.close()`), which certainly was _not_ graceful termination and I saw that the client spewed out an error along the lines of "I tried to send a request but the server didn't respond". 

Also see -- https://github.com/confluentinc/rest-utils/pull/483#discussion_r1621935209